stdlib/REPL/src/REPL.jl

```diff
@@ -1164,9 +1166,25 @@ function setup_interface(
         prompt_suffix = hascolor ?
         (repl.envcolors ? Base.input_color : repl.input_color) : "",
         repl = repl,
-        complete = replc,
+        complete = LineEdit.EmptyCompletionProvider(),
         on_done = respond(repl, julia_prompt) do line
-            # do nothing. Handling is managed by the '\r' keymap
+            end,
+        on_enter = (s) -> begin
+            # ok to do this even if Pkg is loading on the other task because of the loading lock
+            REPLExt = Base.require_stdlib(Pkg_pkgid, "REPLExt")
+            LineEdit.mode(s) === dummy_pkg_mode || return true
```

Comment:
This shouldn't ever be possible to be `false`.

Reply:
Without this a `\r` before Pkg finished loading would cause
```
(......) pkg> st

(@v1.12) pkg> 
```
With it the result is as desired
```
(@v1.12) pkg> st
Status `~/.julia/environments/v1.12/Project.toml`
  [31a5f54b] Debugger v0.7.8
```

I don't quite understand why, tbh.

Reply:
Can you put an `error` there temporarily to see how it's reaching there? I really would have thought it shouldn't be possible.

Reply:
I'm seeing weird behavior. If I add the `error()` it never throws and behavior remains good. If I remove the line the bad behavior returns. `mode(s::MIState)` seems to be read-only, not affecting state, so I don't understand. Note that behavior is always good with `-t1` but typing is blocked, so that's probably not a fair test. Requires `-t2`. It would be helpful to see if you can replicate.

Reply:
I wondered if it could be a race with `REPLExt.__init__()` but I couldn't see anything problematic.

Reply:
See https://github.com/JuliaLang/julia/pull/54730